### PR TITLE
Delete unreferenced packages when deleting a package version

### LINKED ISSUE #20969: Deleted all packages, still failed to delete the account
### Description

> error: Your account owns one or more packages; delete them first.

Gitea asked me to delete the package before deleting my account, and even if I did, it didn't work as it should.

I can reproduce the problem on try.gitea.io

### Gitea Version

1.18.0+dev-346-g94549bcd1

### Can you reproduce the bug on the Gitea demo site?

Yes

### Log Gist

_No response_

### Screenshots

https://user-images.githubusercontent.com/76462613/187001055-eba43c55-0eb4-4cf2-9c2d-41c5f3f93b18.mp4

### Git Version

_No response_

### Operating System

_No response_

### How are you running Gitea?

https://try.gitea.io/

### Database

_No response_

### PR DESCRIPTION
**Pull request by KN4CK3R**
Delete a package if its last version got deleted. Otherwise removing the owner works only after the clean up job ran.

Fix #20969

### CODE REVIEW

File: services/packages/packages.go [hunk 1]
```diff
@@ -340,7 +340,11 @@ func DeletePackageVersionAndReferences(ctx context.Context, pv *packages_model.P
 		}
 	}
 
-	return packages_model.DeleteVersionByID(ctx, pv.ID)
+	if err := packages_model.DeleteVersionByID(ctx, pv.ID); err != nil {
+		return err
+	}
+
+	return deleteUnreferencedPackages(ctx)
```
Comment (zeripath) [inline services/packages/packages.go]: Isn't this is essentially doing a lot of the work that is in cleanup on every delete? i.e. it's tantamount to calling CleanUp for every delete?

Cant we use the information in pv to limit our unreferenced check to that particular package?

Comment (KN4CK3R) [reply services/packages/packages.go]: It's not really much work. In the best case it's 1 sql query, average case is 1+2 queries and the worst case is 1+n*2 queries.

Anyway I just changed the definition of `HasOwnerPackages`, so no additional work to do.

**Comment by lafriks** [comment]
But it would still leave invalid user id referenced as owner?

**Comment by KN4CK3R** [comment]
Yes and they get cleaned up the next time the job runs.

**Review by zeripath — APPROVED** [review]

**Review by lunny — APPROVED** [review]

File: models/packages/package_test.go [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+	owner := unittest.AssertExistsAndLoadBean(t, &user_model.User{ID: 1})
+
+	p, err := packages_model.TryInsertPackage(db.DefaultContext, &packages_model.Package{
+		OwnerID:   owner.ID,
+		LowerName: "package",
+	})
+	assert.NotNil(t, p)
+	assert.NoError(t, err)
+
+	// A package without package versions gets automaticaly cleaned up and should return false
```
Comment (delvh) [inline models/packages/package_test.go]:
```suggestion
	// A package without package versions gets automatically cleaned up and should return false
```

**Review by delvh — APPROVED** [review]

**Comment by zeripath** [comment]
Please send backport

**Pull request merged**